src/Symfony/Bundle/FrameworkBundle/CacheWarmer/SerializerCacheWarmer.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\CacheWarmer;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Symfony\Component\Cache\Adapter\ArrayAdapter;
+use Symfony\Component\Cache\Adapter\PhpArrayAdapter;
+use Symfony\Component\Cache\Adapter\ProxyAdapter;
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+use Symfony\Component\Serializer\Mapping\Factory\CacheClassMetadataFactory;
+use Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactory;
+use Symfony\Component\Serializer\Mapping\Loader\LoaderChain;
+use Symfony\Component\Serializer\Mapping\Loader\LoaderInterface;
+use Symfony\Component\Serializer\Mapping\Loader\XmlFileLoader;
+use Symfony\Component\Serializer\Mapping\Loader\YamlFileLoader;
+
+/**
+ * Warms up XML and YAML serializer metadata.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class SerializerCacheWarmer implements CacheWarmerInterface
+{
+    private $loaders;
+    private $phpArrayFile;
+    private $fallbackPool;
+
+    /**
+     * @param LoaderInterface[]      $loaders      The serializer metadata loaders.
+     * @param string                 $phpArrayFile The PHP file where metadata are cached.
+     * @param CacheItemPoolInterface $fallbackPool The pool where runtime-discovered metadata are cached.
+     */
+    public function __construct(array $loaders, $phpArrayFile, CacheItemPoolInterface $fallbackPool)
+    {
+        $this->loaders = $loaders;
+        $this->phpArrayFile = $phpArrayFile;
+        if (!$fallbackPool instanceof AdapterInterface) {
+            $fallbackPool = new ProxyAdapter($fallbackPool);
+        }
+        $this->fallbackPool = $fallbackPool;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function warmUp($cacheDir)
+    {
+        $adapter = new PhpArrayAdapter($this->phpArrayFile, $this->fallbackPool);
+        $arrayPool = new ArrayAdapter(0, false);
+
+        $metadataFactory = new CacheClassMetadataFactory(
+            new ClassMetadataFactory(new LoaderChain($this->loaders)),
+            $arrayPool
+        );
+
+        foreach ($this->extractSupportedLoaders($this->loaders) as $loader) {
+            foreach ($loader->getMappedClasses() as $mappedClass) {
+                $metadataFactory->getMetadataFor($mappedClass);
+            }
+        }
+
+        $values = $arrayPool->getValues();
+        $adapter->warmUp($values);
+
+        foreach ($values as $k => $v) {
```

Comment:
This is not planned and not done anywhere else, and I think when this will happen we will have a lot of work to adapt the whole codebase anyway. I don't think it has much added value, don't you think?